crates/biome_rowan/src/green/node.rs

```diff
@@ -232,8 +232,8 @@ impl ops::Deref for GreenNode {
     fn deref(&self) -> &GreenNodeData {
         unsafe {
             let repr: &Repr = &self.ptr;
-            let repr: &ReprThin = &*(repr as *const Repr as *const ReprThin);
-            mem::transmute::<&ReprThin, &GreenNodeData>(repr)
+            let repr: &ReprThin = &*(repr as *const Repr).cast();
+            &*((repr as *const HeaderSlice<GreenNodeHead, _>).cast())
```

Comment:
Would it be possible to avoid changing too much rowan?

Reply:
ARe there specific reasons to avoid to change rowan? Because most of our crates depend on?

Reply:
Mainly because it's a fork, and we depend on it very much. We already changed some stuff, true, but the core is mainly unchanged. In case things change upstream, we can easily identify the common paths.